allennlp_models/common/model_card.py

```diff
@@ -0,0 +1,282 @@
+import os
+import logging
+from dataclasses import dataclass
+from typing import Optional, Union, Dict, Any
+from allennlp.common.from_params import FromParams
+
+from allennlp.models import Model
+from allennlp.common.checks import ConfigurationError
+
+logger = logging.getLogger(__name__)
+
+
+def get_description(model_class):
+    """
+    Returns the model's description from the docstring.
+    """
+    return model_class.__doc__.split("# Parameters")[0].strip()
+
+
+class ModelCardInfo(FromParams):
+    def to_dict(self):
+        """
+        Only the non-empty attributes are returned, to minimize empty values.
+        """
+        info = {}
+        for key, val in self.__dict__.items():
+            if val:
+                info[key] = val
+        return info
+
+    def __str__(self):
+        display = ""
+        for key, val in self.to_dict().items():
+            display += "\n" + key.replace("_", " ").capitalize() + ": "
+            display += "\n\t" + val.replace("\n", "\n\t") + "\n"
```

Comment:
This will mean that the final string starts and ends with `"\n"`. Return `display.strip()`?